api/src/main/java/com/valtech/aem/saas/api/indexing/PathExternalizer.java

```diff
@@ -0,0 +1,34 @@
+package com.valtech.aem.saas.api.indexing;
+
+import java.util.List;
+import java.util.stream.Collectors;
+
+/**
+ * Service interface for externalizing resource path.
+ */
+public interface PathExternalizer {
+
+  /**
+   * Gets the rank of the externalizer, which determines its order in the pipeline. Rank must be unique for each
+   * implementation. If there are two or more implementations with same rank, only the one will be registered in the
+   * pipeline. The others will be ignored.
+   *
+   * @return rank value.
+   */
+  int getRank();
```

Comment:
- [ ] let's try to use OSGi service ranking instead. Please timebox this to 1h. Otherwise we will keep current solution